lib/ocpp/common/websocket/websocket_libwebsockets.cpp

```diff
@@ -604,429 +678,411 @@ void WebsocketLibwebsockets::client_loop() {
+                local_data->update_state(EConnectionState::RECONNECTING);
+                reconnect_delay = this->get_reconnect_interval();
+                try_reconnect = true;
+
+                // Increment reconn attempts
+                this->connection_attempts += 1;
+
+                EVLOG_info << "Connection not successful, attempting internal reconnect in: " << reconnect_delay
+                           << "ms";
+            } else {
+                local_data->update_state(EConnectionState::FINALIZED);
+                try_reconnect = false;
+
+                EVLOG_info << "Connection reconnect attempts exhausted, exiting websocket loop, passing back control "
+                              "to the application logic";
             }
-            if (!message_queue_empty) {
-                lws_callback_on_writable(local_data->get_conn());
+        }
+
+        // Wait until new connection attempt
+        if (local_data->get_state() == EConnectionState::RECONNECTING) {
+            auto end_time = std::chrono::steady_clock::now() + std::chrono::milliseconds(reconnect_delay);
+
+            while ((std::chrono::steady_clock::now() < end_time) && (false == local_data->is_interupted())) {
+                std::this_thread::sleep_for(std::chrono::milliseconds(100));
+            }
+
+            if (true == local_data->is_interupted()) {
+                try_reconnect = false;
+                EVLOG_info << "Interrupred reconnect attempt, not reconnecting!";
+            } else {
+                EVLOG_info << "Attempting reconnect after a wait of: " << reconnect_delay << "ms";
             }
         }
+    } while (try_reconnect); // End trying to connect
+
+    // Give back control to the application
+    if (false == local_data->is_stop_executed()) {
+        this->push_deferred_callback([this]() {
+            if (this->stopped_connecting_callback) {
+                this->stopped_connecting_callback(WebsocketCloseReason::Normal);
+            } else {
+                EVLOG_error << "Stopped connecting callback not registered!";
+            }
 
-        // Client loop finished for our tid
-        EVLOG_debug << "Exit client loop with ID: " << std::hex << std::this_thread::get_id();
+            if (this->disconnected_callback) {
+                this->disconnected_callback();
+            } else {
+                EVLOG_error << "Disconnected callback not registered!";
+            }
+        });
     }
+
+    // Client loop finished for our tid
+    EVLOG_info << "Exit websocket client loop with ID: " << std::hex << std::this_thread::get_id();
 }
 
-// Will be called from external threads as well
-bool WebsocketLibwebsockets::connect() {
-    if (!this->initialized()) {
-        return false;
-    }
+void WebsocketLibwebsockets::clear_all_queues() {
+    message_queue.clear();
+    recv_buffered_message.clear();
+    recv_message_queue.clear();
```

Comment:
```suggestion
    this->message_queue.clear();
    this->recv_buffered_message.clear();
    this->recv_message_queue.clear();
```

Looks like we don't do this throughout the whole file anyway.